src/hdf5_time_series_storage.jl

```diff
@@ -182,43 +200,78 @@ function remove_time_series!(
     end
 end
 
+function _make_time_array(timestamps, data, columns::Union{Vector{Symbol}, Nothing})
+    if columns === nothing
+        return TimeSeries.TimeArray(timestamps, data)
+    else
+        return TimeSeries.TimeArray(timestamps, data, columns)
+    end
+end
+
 function get_time_series(
     storage::Hdf5TimeSeriesStorage,
-    uuid::UUIDs.UUID;
-    index = 0,
-    len = 0,
-)::TimeSeries.TimeArray
+    uuid::UUIDs.UUID,
+    index::Int,
+    len::Int,
+    count::Int,
+)::DataStructures.SortedDict{Dates.DateTime, TimeSeries.TimeArray}
     return HDF5.h5open(storage.file_path, "r") do file
         root = _get_root(storage, file)
         path = _get_time_series_path(root, uuid)
-
-        if index == 0
-            data = HDF5.read(path["data"])
-            timestamps = HDF5.read(path["timestamps"])
+        _initial_time_stamp = HDF5.read(HDF5.attrs(path)["initial_time"])
+        initial_time_stamp = Dates.epochms2datetime(_initial_time_stamp)
+        resolution = Dates.Millisecond(HDF5.read(HDF5.attrs(path)["resolution"]))
+        series_length = HDF5.read(HDF5.attrs(path)["length"])
```

Comment:
This information is already provided by HDF and so we don't need to store it.  It can be retrieved with `size(path["data"])`.

Reply:
move to size. Leave comments relating to the meta data names